rts/Sim/Weapons/Cannon.cpp

```diff
@@ -62,26 +62,24 @@ bool CCannon::HaveFreeLineOfFire(const float3 srcPos, const float3 tgtPos, const
 	if (projectileSpeed == 0.0f)
 		return true;
 
-	float3 launchDir = CalcWantedDir(tgtPos - srcPos);
-	float3 targetVec = (tgtPos - srcPos) * XZVector;
+	float3 launchDir = CalcWantedDir(tgtPos - weaponMuzzlePos);
```

Comment:
Is `weaponMuzzlePos` not equal to `srcPos`? If so, what does `srcPos` represent?